docs/developers-guide/driver-changelog.md

```diff
@@ -58,6 +58,10 @@ title: Driver interface changelog
   clause (equivalent of SQL `lead` and `lag` functions). This is enabled by default for `:sql` and `:sql-jdbc`-based
   drivers. Other drivers should add an implementation for this clause and enable the feature flag.
 
+- `metabase.driver/field-values-compatible?` multimethod was added to enable drivers to opt out from field values
+  computations for specific fields. Changed was motivated by the Druid JDBC's inability to execute field value queries
+  from its `COMPLEX<...>` type columns.
```

Comment:
Being a little more detailed here for what third-party driver authors need to do would be helpful, I would mention how by default this is enabled for all fields and driver authors shouldn't need to do anything unless they want to disable field values scanning for a specific Field.

Reply:
I'll keep in mind describing the defaults in the future. Thanks.